core/src/network/reducers/publish.rs

```diff
@@ -43,6 +46,38 @@ fn publish_entry(
     )
 }
 
+/// Send to network a request to publish a header entry alone
+/// This is similar to publishing a regular entry but it is its own special dummy header.
+fn publish_header(
+    network_state: &mut NetworkState,
+    root_state: &State,
+    chain_header: &ChainHeader,
+) -> Result<(), HolochainError> {
+    let header_entry = Entry::ChainHeader(chain_header.clone());
+    let header_entry_header = create_new_chain_header(
+        &header_entry,
+        &root_state.agent(),
+        &StateWrapper::from(root_state.clone()),
```

Comment:
@lucksus This is one thing I am not sure about. Can I create a `&StateWrapper` from a `&State` without cloning it? Also I am not sure if it is ok to clone the state like this...

Reply:
Great idea. This is done now